### PR TITLE
feat(GiniHealthSDK): Download pdf API v4 

### PR DESCRIPTION
**Pull request by razvancapra**
Tasks included:
https://ginis.atlassian.net/browse/IPC-252
https://ginis.atlassian.net/browse/IPC-267
https://ginis.atlassian.net/browse/IPC-259

### CODE REVIEW

File: HealthAPILibrary/GiniHealthAPILibrary/Sources/GiniHealthAPILibrary/Documents/APIResource.swift [hunk 1]
```diff
@@ -46,9 +46,9 @@ struct APIResource<T: Decodable>: Resource {
     
     var apiVersion: Int {
         switch domain {
-        case .default: return 3
+        case .default: return 4
         case .gym: return 2
```
Comment (zladzeyka) [inline HealthAPILibrary/GiniHealthAPILibrary/Sources/GiniHealthAPILibrary/Documents/APIResource.swift:50]: [Request]: @razvancapra we need to remove `gym` from `HealthAPILibrary` and `HealthSDK` resources

**Review by zladzeyka — CHANGES_REQUESTED** [review]
👏🏻 Great job @razvancapra 
there're a couple of suggestion. I will continue with testing the PR

File: HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentComponent/PaymentComponentViewModel.swift [hunk 2]
```diff
@@ -52,7 +52,7 @@ extension PaymentComponentViewProtocol {
 }
 
 final class PaymentComponentViewModel {
-    let giniHealthConfiguration = GiniHealthConfiguration.shared
+    let giniHealthConfiguration: GiniHealthConfiguration
```
[the remark this answers is not in the document]
Comment (razvancapra) [reply HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentComponent/PaymentComponentViewModel.swift:55]: I got some memory warnings into previous implementation due to the high number of cells that where getting instantiated while quick scrolling. I decided to inject the configuration in there.

File: HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Resources/GiniImages.xcassets/more_vertical.imageset/more-vertical (1).pdf
[the remark this answers is not in the document]
Comment (razvancapra) [reply HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Resources/GiniImages.xcassets/more_vertical.imageset/more-vertical (1).pdf:1]: Yes. We do need. One of them is for Dark more, the other one is for light mode.

**Pull request merged**